src/RetryMiddleware.php

```diff
@@ -80,6 +87,18 @@ public static function createDefaultDecider($maxRetries = 3)
                 return true;
             } elseif (isset(self::$retryStatusCodes[$error->getStatusCode()])) {
                 return true;
+            } elseif (($previous = $error->getPrevious())
+                && $previous instanceof ConnectException) {
+                if ($version[0] === '6') {
```

Comment:
Rather than checking the version, couldn't you check if the `getHandlerContext` method exists?